src/pybullet_industrial/g_code_processor.py

```diff
@@ -0,0 +1,405 @@
+import time
+import pybullet as p
+from pybullet_industrial import RobotBase
+from pybullet_industrial import ToolPath
+from pybullet_industrial import linear_interpolation
+from pybullet_industrial import circular_interpolation
+import numpy as np
+
+
+class GCodeProcessor:
```

Comment:
@liquidcronos many thanks for the thorough code review! I changed the name of the class for following the CamelCase naming convetions, which is why the old comments of the gcode_class are not displayed on this class.
In the new class GCodeProcessor I implemented all the proposed changes, besides the parts I created a new comment.